testsuite/classlibrary/TestIOUGens.sc

```diff
@@ -0,0 +1,127 @@
+TestIOUGens : UnitTest {
+    var server;
+
+	setUp {
+		server = Server(this.class.name);
+		server.bootSync;
+	}
+
+	tearDown {
+		server.quit;
+		server.remove;
+	}
+
+    //Test that InFeedback is equal to In when write / read are in the right order
+    test_InFeedback_equals_In {
+        var condition = Condition();
+        var bus = Bus.audio(server);
+
+        {
+            var writer = Out.ar(bus.index, SinOsc.ar(1));
+            var in = In.ar(bus.index);
+            var inFeedback = InFeedback.ar(bus.index);
+            inFeedback - in;
+        }.loadToFloatArray(0.5, server, { | data |
+            this.assertFloatEquals(data.sum, 0.0, "InFeedback is not equal to In", report:true);
+            condition.unhang;
+        });
+
+        condition.hang;
+    }
+
+    //Test that AudioControl is equal to In when write / read are in the right order
+    test_AudioControl_equals_In {
+        var condition = Condition();
+        var bus = Bus.audio(server);
+        var sumBus = Bus.audio(server);
+
+        var writer = {
+            Out.ar(bus.index, SinOsc.ar(1));
+            Silent.ar;
+        };
+
+        var in = {
+            Out.ar(sumBus.index, In.ar(bus.index));
+            Silent.ar;
+        };
+
+        var audioControl = {
+            Out.ar(sumBus.index, -1 * (\in.ar)); //Invert to perform difference at sumBus
+            Silent.ar;
+        };
+
+        var sum = {
+            In.ar(sumBus.index);
+        };
+
+        server.bind({
+            sum.loadToFloatArray(0.5, server, { | data |
+                this.assertFloatEquals(data.sum, 0.0, "AudioControl is not equal to In", report:true);
```

Comment:
I'd suggest the message to say: "AudioControl should be equal to In for read after write".